module/spl/spl-zlib.c

```diff
@@ -202,7 +202,7 @@ spl_zlib_init(void)
 	zlib_workspace_cache = kmem_cache_create(
 	    "spl_zlib_workspace_cache",
 	    size, 0, NULL, NULL, NULL, NULL, NULL,
-	    KMC_VMEM | KMC_NOEMERGENCY);
+	    KMC_KVMEM | KMC_NOEMERGENCY);
```

Comment:
It is a pre-existing issue, but this risks deadlocking the system in direct reclaim inside the zio pipeline during low memory conditions on pre-3.18 kernels.

Reply:
My remark turns out to be incorrect. Surprisingly, KMC_NOEMERGENCY is not actually implemented, so there is no risk of a deadlock. We should drop KMC_NOEMERGENCY here.

Reply:
I have opened #9119 to remove KMC_NOEMERGENCY.

Reply:
your assumption regarding kmalloc/zstd/zio is invalid, too, then, isn‘t it?

Reply:
That is correct, but my remark "spl_kmem_alloc_max wouldn't apply to `KM_VMEM` if it is being redefined to mean kvmalloc()" should still apply. spl_kmem_alloc_max never applied to `KMC_VMEM`. I don't think the reasoning of keeping this separate to avoid `spl_kmem_alloc_max` is correct.